javax-money/src/test/java/com/fasterxml/jackson/datatype/money/MonetaryAmountDeserializerTest.java

```diff
@@ -0,0 +1,273 @@
+        final MonetaryAmount amount = unit.readValue(content, type);
+
+        assertThat(amount.getNumber().numberValueExact(BigDecimal.class)).isEqualTo((new BigDecimal("29.95")));
+        assertThat(amount.getCurrency().getCurrencyCode()).isEqualTo(("EUR"));
+    }
+
+    @Test
+    @Parameters(method = "data")
+    public void shouldIgnoreFormattedValue(final Class<M> type, final Configurer configurer) throws IOException {
+        final ObjectMapper unit = unit(configurer);
+
+        final String content = "{\"amount\":29.95,\"currency\":\"EUR\",\"formatted\":\"30.00 EUR\"}";
+        final MonetaryAmount amount = unit.readValue(content, type);
+
+        assertThat(amount.getNumber().numberValueExact(BigDecimal.class)).isEqualTo((new BigDecimal("29.95")));
+        assertThat(amount.getCurrency().getCurrencyCode()).isEqualTo(("EUR"));
+    }
+
+    @Test
+    @Parameters(method = "data")
+    public void shouldUpdateExistingValueUsingTreeTraversingParser(final Class<M> type,
+            final Configurer configurer) throws IOException {
+        final ObjectMapper unit = unit(configurer);
+
+        final String content = "{\"amount\":29.95,\"currency\":\"EUR\"}";
+        final MonetaryAmount amount = unit.readValue(content, type);
+
+        assertThat(amount).isNotNull();
+
+        // we need a json node to get a TreeTraversingParser with codec of type ObjectReader
+        final JsonNode ownerNode =
+                unit.readTree("{\"value\":{\"amount\":29.95,\"currency\":\"EUR\",\"formatted\":\"30.00EUR\"}}");
+
+        final Owner owner = new Owner();
+        owner.setValue(amount);
+
+        // try to update
+        final Owner result = unit.readerForUpdating(owner).readValue(ownerNode);
+        assertThat(result).isNotNull();
+        assertThat(result.getValue()).isEqualTo((amount));
+    }
+
+    private static class Owner {
+
+        private MonetaryAmount value;
+
+        MonetaryAmount getValue() {
+            return value;
+        }
+
+        void setValue(final MonetaryAmount value) {
+            this.value = value;
+        }
+
+    }
+
+    @Test
+    @Parameters(method = "data")
+    public void shouldFailToDeserializeWithoutAmount(final Class<M> type, final Configurer configurer) {
+        final ObjectMapper unit = unit(configurer);
+
+        final String content = "{\"currency\":\"EUR\"}";
+
+        final JsonProcessingException exception = assertThrows(
+                JsonProcessingException.class, () -> unit.readValue(content, type));
+
+        assertThat(exception.getMessage()).contains("Missing property: 'amount'");
+    }
+
+    @Test
+    @Parameters(method = "data")
+    public void shouldFailToDeserializeWithoutCurrency(final Class<M> type, final Configurer configurer) {
+        final ObjectMapper unit = unit(configurer);
+
+        final String content = "{\"amount\":29.95}";
+
+        final JsonProcessingException exception = assertThrows(
```

Comment:
While technically correct, use of `JsonProcessingException` is not optimal as we should check specific subtype that is being thrown, JPE is the most generic Jackson exception base class.